src/gui/screen_menu_filament.cpp

```diff
@@ -2,50 +2,174 @@
 
 #include "gui.h"
 #include "screen_menu.hpp"
+#include "WindowMenuItems.hpp"
 #include "filament.h"
 #include "filament_sensor.h"
 #include "marlin_client.h"
 #include "menu_vars.h"
 #include "window_dlg_load_unload.h"
 #include "screens.h"
 #include "dbg.h"
-#include "status_footer.h"
 #include "DialogHandler.hpp"
 
-enum { F_EEPROM = 0x01,
-    F_SENSED = 0x02 };
-
-typedef enum {
-    MI_RETURN,
-    MI_LOAD,
-    MI_UNLOAD,
-    MI_CHANGE,
-    MI_PURGE,
-    MI_COUNT
-} MI_t;
-
-const menu_item_t _menu_filament_items[] = {
-    { { "Load Filament", 0, WI_LABEL }, SCREEN_MENU_NO_SCREEN },
-    { { "Unload Filament", 0, WI_LABEL }, SCREEN_MENU_NO_SCREEN },
-    { { "Change Filament", 0, WI_LABEL }, SCREEN_MENU_NO_SCREEN },
-    { { "Purge Filament", 0, WI_LABEL }, SCREEN_MENU_NO_SCREEN },
+/// Sets temperature of nozzle not to ooze before print (MBL)
+void setPreheatTemp() {
+    marlin_gcode_printf("M104 S%d D%d", (int)PREHEAT_TEMP, (int)filaments[get_filament()].nozzle);
```

Comment:
better than gcode is using marlin_set_var, but i am not sure if it is implemented for the second parameter of M104.